### ISSUE TITLE
Docs: add type annotation for `tseslint.config()` usage examples

### ISSUE THREAD

**Issue by MonstraG** [typescript-eslint/typescript-eslint#10508]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

In examples of `eslint.config.mjs`, add:
```
// @ts-check

import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';

+ /** @type {FlatConfig.ConfigArray} */
export default tseslint.config(
  eslint.configs.recommended,
  tseslint.configs.recommended,
);
```
(or whatever there actually should be)

(or add this as a separate FAQ item or something)

### Affected URL(s)

https://typescript-eslint.io/getting-started/

### Additional Info

I request this for the following 2 reasons:
1) My repositories use types as much as possible, and whenever I have to write .mjs files, I always add `type` there, for typescript to check them also
2) without appropriate type annotation, typescript reports;
```
TS2742: The inferred type of config cannot be named without a reference to
.pnpm/@typescript-eslint+utils@8.18.0_eslint@9.17.0_typescript@5.7.2/node_modules/@typescript-eslint/utils/ts-eslint
. This is likely not portable. A type annotation is necessary.
```

While the error sounds important, I would still be raising this issue even if error was not there.

Additionally, I don't actually know what to write there, and never could I find this information. I searched for `jsdoc`, `type annotations`, `ConfigArray` and `portable` but was unable to find any issues that would give me even a hint at what should be the correct `@type`.

**Comment by bradzacher** [comment]
Could you please provide a reproduction for the error you're seeing?

**Comment by MonstraG** [comment]
https://github.com/MonstraG/tslintreporoerror

![image](https://github.com/user-attachments/assets/1463a446-32fb-4e91-b6a4-cb87e8905908)

**Comment by MonstraG** [comment]
it seeeeeems that for writing:
```js

/** @type {import("typescript-eslint/dist/config-helper").InfiniteDepthConfigWithExtends} */
const a = {
    files: [],
    rules: {},
}

/** @type {import("typescript-eslint/dist/config-helper").InfiniteDepthConfigWithExtends} */
const b = {
    files: [],
    rules: {},
}

export const appConfig = tseslint.config(a, b);
```

could work, but of course that's from `dist` and not actually exported and doesn't build.

**Comment by bradzacher** [comment]
Your `eslint.config.mjs` is covered by a tsconfig that has `composite: true`. This implies `declaration: true` -- so your config is trying to generate a `.d.ts` file for a JS config file.

This is a highly unusual setup and not recommended for obvious reasons and is why you're getting the error. TS is trying to generate a `.d.ts` file and it's not able to name the types because we hadn't exported all of the types from the top-level, and TS does not like inserting imports to packages you haven't imported explicitly.

We can fix this on our side so that it "just works"

**Comment by JoshuaKGoldberg** [comment]
Wow, what timing! I _just_ hit this on my own monorepo with a `composite: true` TSConfig used for `eslint.config.js` type info. +1
